src/value.h

```diff
@@ -20,41 +20,43 @@
  * differing types is used to maintain the Cypher-defined global sort order
  * in the SIValue_Order routine. */
 typedef enum {
-	T_MAP = (1 << 0),
-	T_NODE = (1 << 1),
-	T_EDGE = (1 << 2),
-	T_ARRAY = (1 << 3),
-	T_PATH = (1 << 4),
-	T_DATETIME = (1 << 5),
+	T_MAP           = (1 << 0),
+	T_NODE          = (1 << 1),
+	T_EDGE          = (1 << 2),
+	T_ARRAY         = (1 << 3),
+	T_PATH          = (1 << 4),
+	T_DATETIME      = (1 << 5),
 	T_LOCALDATETIME = (1 << 6),
-	T_DATE = (1 << 7),
-	T_TIME = (1 << 8),
-	T_LOCALTIME = (1 << 9),
-	T_DURATION = (1 << 10),
-	T_STRING = (1 << 11),
-	T_BOOL = (1 << 12),   // shares 'longval' representation in SIValue union
-	T_INT64 = (1 << 13),
-	T_DOUBLE = (1 << 14),
-	T_NULL = (1 << 15),
-	T_PTR = (1 << 16),
-	T_POINT = (1 << 17),  // TODO: verify type order of point
-	T_VECTOR_F32 = (1 << 18),
+	T_DATE          = (1 << 7),
+	T_TIME          = (1 << 8),
+	T_LOCALTIME     = (1 << 9),
+	T_DURATION      = (1 << 10),
+	T_STRING        = (1 << 11),
+	T_BOOL          = (1 << 12),  // shares 'longval' representation in SIValue  union
+	T_INT64         = (1 << 13),
+	T_DOUBLE        = (1 << 14),
+	T_NULL          = (1 << 15),
+	T_PTR           = (1 << 16),
+	T_POINT         = (1 << 17),
+	T_VECTOR_F32    = (1 << 18),
+	T_INTERN_STRING = (1 << 19),
```

Comment:
rename to T_INTERN = (1 << 64)
and add a #define T_INTERN_STRING T_STRING | T_INTERN

Reply:
Currently the largest enum value in this type is `T_INTERN_STRING`
Which means the `SIType` can fit within a 32bit number and indeed `sizeof(SIType) = 4`
If we were to introduce `T_INTERN = (1 << 63)` then `SIType` will double in size from 4 bytes to 8 bytes, bloating all of our SIValues. For that reason and the unlikelihood that we would intern other datatypes I'm inclined to leave things as they are.

Reply:
I've decided to go with:
`T_INTERN_STRING = (1 << 19) | T_STRING,` which gives us the benefits of both worlds.
if we need another interned type we can use: `T_INTERN_MAP = (1 << 19) | T_MAP,`